src/views/collection-view/server.ts

```diff
@@ -112,37 +110,40 @@ export async function getCollectionPageProps(
 	}
 }
 
-export async function getCollectionPaths(
-	product: ProductOption
-): Promise<CollectionPagePath[]> {
-	const collections = await cachedGetAllCollections({
-		product: { slug: product },
-	})
+async function getCollectionPagePaths(): Promise<CollectionPagePath[]> {
+	const collections = await cachedGetAllCollections()
 	// Only build collections where this product is the main 'theme'
 	// @TODO once we implement the `theme` query option, remove the theme filtering
 	// https://app.asana.com/0/1201903760348480/1201932088801131/f
-	const filteredCollections = collections.filter((c) => c.theme === product)
-	const paths = filteredCollections.map((collection) => ({
-		params: {
-			collectionSlug: splitProductFromFilename(collection.slug),
-		},
-	}))
+	const filteredCollections = collections.filter((c) =>
+		getIsBetaProduct(c.theme as LearnProductSlug)
+	)
+	const paths = filteredCollections.map((collection) => {
+		// assuming slug structure of :product/:filename
+		const [product, filename] = collection.slug.split('/')
+		return {
+			params: {
+				product,
+				collectionSlug: filename,
+			},
+		}
+	})
```

Comment:
📌 Out of scope of this PR since it was already present, but do we need separate `.filter` and `.map` calls here, or could we build a new array and just iterate of `collections` once using `forEach`? 💭

Reply:
I made some updates with your suggestions!

Reply:
Oh awesome! Appreciate knocking this out early here, and looks great. 🤩